sklearn/preprocessing/tests/test_imputation.py

```diff
@@ -364,3 +392,144 @@ def test_imputation_copy():
 
     # Note: If X is sparse and if missing_values=0, then a (dense) copy of X is
     # made, even if copy=False.
+
+
+def test_mice_rank_one():
+    d = 100
+    A = np.random.random((d, 1))
+    B = np.random.random((1, d))
+    X = np.dot(A, B)
+    nan_mask = np.random.random((d, d)) < 0.5
+    X_missing = X.copy()
+    X_missing[nan_mask] = np.nan
+
+    imputer = MICEImputer(n_imputations=5,
+                          n_burn_in=5,
+                          verbose=True)
+    X_filled = imputer.fit_transform(X_missing)
+    assert_array_almost_equal(X_filled, X, decimal=2)
+
+
+def test_mice_imputation_order():
+    n = 100
+    d = 10
+    X = sparse_random_matrix(n, d, density=0.10).toarray()
+    X[:, 0] = 1  # this column shouldn't be ever used
+
+    for imputation_order in ['random', 'roman', 'monotone',
+                             'revmonotone', 'arabic']:
+        imputer = MICEImputer(missing_values=0,
+                              n_imputations=1,
+                              n_burn_in=1,
+                              n_nearest_features=5,
+                              min_value=0,
+                              max_value=1,
+                              verbose=False,
+                              imputation_order=imputation_order)
+        imputer.fit_transform(X)
+        ordered_idx = [i.feat_idx for i in imputer.imputation_sequence_]
+        if imputation_order == 'roman':
+            assert np.all(ordered_idx[:d-1] == np.arange(1, d))
+        elif imputation_order == 'arabic':
+            assert np.all(ordered_idx[:d-1] == np.arange(d-1, 0, -1))
+        elif imputation_order == 'random':
```

Comment:
Please check the length of the monotones

Reply:
Done.